app/assets/javascripts/discourse/widgets/quick-access-messages.js.es6

```diff
@@ -0,0 +1,59 @@
+import DiscourseURL from "discourse/lib/url";
+import QuickAccessPanel from "discourse/widgets/quick-access-panel";
+import { createWidgetFrom } from "discourse/widgets/widget";
+import { postUrl } from "discourse/lib/utilities";
+
+const ICON = "notification.private_message";
+
+let staleItems = [];
+
+function toItem(message) {
+  const lastReadPostNumber = message.last_read_post_number || 0;
+  const nextUnreadPostNumber = Math.min(
+    lastReadPostNumber + 1,
+    message.highest_post_number
+  );
+
+  return Ember.Object.create({
```

Comment:
Does this need to be an `Ember.Object`? They are heavier than regular JS objects, and if you don't need to observe or use computed properties on them I recommend the plain JS version.